From efc67121c6be96424c876151a928a4d7216a36e8 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:48 -0400
Subject: [usb] xhci: Limit the spurious wakeup fix only to HP machines

Message-id: <1410887463-20674-217-git-send-email-dzickus@redhat.com>
Patchwork-id: 95227
O-Subject: [RHEL7 PATCH 216/591] xhci: Limit the spurious wakeup fix only to HP machines
Bugzilla: 1069461
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1069461

commit 6962d914f317b119e0db7189199b21ec77a4b3e0
Author: Takashi Iwai <tiwai@suse.de>
Date:   Mon Dec 9 14:53:36 2013 +0100

    xhci: Limit the spurious wakeup fix only to HP machines

    We've got regression reports that my previous fix for spurious wakeups
    after S5 on HP Haswell machines leads to the automatic reboot at
    shutdown on some machines.  It turned out that the fix for one side
    triggers another BIOS bug in other side.  So, it's exclusive.

    Since the original S5 wakeups have been confirmed only on HP machines,
    it'd be safer to apply it only to limited machines.  As a wild guess,
    limiting to machines with HP PCI SSID should suffice.

    This patch should be backported to kernels as old as 3.12, that
    contain the commit 638298dc66ea36623dbc2757a24fc2c4ab41b016 "xhci: Fix
    spurious wakeups after S5 on Haswell".

    Bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=66171
    Cc: stable@vger.kernel.org
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
    Tested-by: <dashing.meng@gmail.com>
    Reported-by: Niklas Schnelle <niklas@komani.de>
    Reported-by: Giorgos <ganastasiouGR@gmail.com>
    Reported-by: <art1@vhex.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index b8dffd5..73f5208 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -128,7 +128,12 @@ static void xhci_pci_quirks(struct device *dev, struct xhci_hcd *xhci)
    * any other sleep) on Haswell machines with LPT and LPT-LP
    * with the new Intel BIOS
    */
-  xhci->quirks |= XHCI_SPURIOUS_WAKEUP;
+  /* Limit the quirk to only known vendors, as this triggers
+   * yet another BIOS bug on some other machines
+   * https://bugzilla.kernel.org/show_bug.cgi?id=66171
+   */
+  if (pdev->subsystem_vendor == PCI_VENDOR_ID_HP)
+   xhci->quirks |= XHCI_SPURIOUS_WAKEUP;
  }
  if (pdev->vendor == PCI_VENDOR_ID_ETRON &&
    pdev->device == PCI_DEVICE_ID_ASROCK_P67) {
-- 
1.7.1